components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java

```diff
@@ -662,23 +654,34 @@ public void completeWorkflow(WorkflowExecutor workflowExecutor, Workflow workflo
         } else {
             String message = "Invalid workflow type:  " + workflow.getWorkflowType();
             log.error(message);
-            throw new WorkflowException(message);
+            throw new APIManagementException(message, ExceptionCodes.WORKFLOW_EXCEPTION);
         }
     }
     
-    private void updateWorkflowEntries(Workflow workflow) throws APIMgtDAOException {
+    private void updateWorkflowEntries(Workflow workflow) throws APIManagementException {
         workflow.setUpdatedTime(LocalDateTime.now());
-        getWorkflowDAO().updateWorkflowStatus(workflow);
-        //TODO stats stuff
+        try {
+            getWorkflowDAO().updateWorkflowStatus(workflow);
+            //TODO stats stuff
+        } catch (APIMgtDAOException e) {
+            String message = "Error while updating workflow entry";
```

Comment:
Shall we add some context information to the error log so that we can narrow down the error when we are debugging other than just mentioning  "Error while updating workflow entry". May be the blow sample code will be useful.

"Error while updating workflow entry for API:" + workflow.getApiName(); + ":" + workflow.getApiVersion();